### ISSUE TITLE
When splitting audiobook file, the previous/next chapter is included in the split file 

### ISSUE THREAD

**Issue by ZLoth** [openaudible/openaudible#1461]
This was discovered in relation to Audio Book Shelf issue [[Bug]: When importing audiobook as multiple MP3/M4B files, chapter titles are duplicated with same time stamp #3361](https://github.com/advplyr/audiobookshelf/issues/3361). When splitting the audiobook into multiple files, each file not only includes the current chapter information, but also the previous and next chapters as almost zero-length chapters. 

STEPS TO REPLICATE:
1. Download a Audible book file
2. Split the audiobook
3. Check the metadata contents

WHAT SHOULD HAPPEN: The metadata should only have the current chapter.
WHAT REALLY HAPPENS: The metadata includes the current chapter at the proper length, but also the previous and next chapters as almost zero length data.

**Comment by openaudible** [comment]
If you look at one of the split files does ffprobe show the zero length chapters?  Feel free to send a sample file that shows the bug to openaudible.org/check

**Comment by ZLoth** [comment]
> If you look at one of the split files does ffprobe show the zero length chapters? Feel free to send a sample file that shows the bug to openaudible.org/test

404 error
![image](https://github.com/user-attachments/assets/8200ee50-5376-4135-8222-71a6a87949ab)

**Comment by openaudible** [comment]
Oops. 

https://openaudible.org/check

**Comment by ZLoth** [comment]
I have uploaded a test chapter to that URL. Here is the relevant information as detected by AudioBookShelf:

```
  "chapters": [
    {
      "id": 0,
      "time_base": "1/1000",
      "start": 0,
      "start_time": "0.000000",
      "end": 0,
      "end_time": "0.000000",
      "tags": {
        "title": "Part I"
      }
    },
    {
      "id": 1,
      "time_base": "1/1000",
      "start": 0,
      "start_time": "0.000000",
      "end": 906562,
      "end_time": "906.562000",
      "tags": {
        "title": "Chapter I"
      }
    },
    {
      "id": 2,
      "time_base": "1/10000000",
      "start": 9065620000,
      "start_time": "906.562000",
      "end": 9066078910,
      "end_time": "906.607891",
      "tags": {
        "title": "Chapter II"
      }
    }
  ],
```

[PROBE - 03 - The Adventures of Tom Sawyer (TEST) - Chapter I.txt](https://github.com/user-attachments/files/17137713/PROBE.-.03.-.The.Adventures.of.Tom.Sawyer.TEST.-.Chapter.I.txt)

**Comment by openaudible** [comment]
Confirmed. Working on a fix. 
I will include just the one chapter, with the corrected offsets. 
For split by time, I will include all (partial and full) chapters.

**Comment by ZLoth** [comment]
> Confirmed. Working on a fix. I will include just the one chapter, with the corrected offsets. For split by time, I will include all (partial and full) chapters.

Thank you. Looking forward to testing this in beta.

**Comment by openaudible** [comment]
First beta with update: 
https://openaudible.org/beta

**Comment by ZLoth** [comment]
Still not working with the latest beta. I just tested with ASIN B00P1NCF6S - [Good to Go - The Life and times of a Decorated Member of the U.S. Navy's Elite Seal Team Two](https://www.audible.com/pd/Good-to-Go-Audiobook/B00P1NCF6S). I found a hidden mode in MP3TAG (File → Options → Tags → Advanced → List chapters as separate files), and when I open up the single M4B file, I see the following:

![image](https://github.com/user-attachments/assets/35df7443-a8d8-498f-b278-c701b7a1a8d0)

However, when splitting that book, I see the following when I open up all of the resulting M4B files:

![image](https://github.com/user-attachments/assets/a4c5a477-9882-447d-aa81-ca7f9be6811f)

Or, when I look at a single chapter (In this case 02 - Good to Go - Prologue.m4b): 
![image](https://github.com/user-attachments/assets/8f0e8214-2fb9-4aa1-8d2f-560d59b010b0)

When I use the "Probe" tool in ABS for that same chapter, I see the following in lines 152 to 186:
```
  "chapters": [
    {
      "id": 0,
      "time_base": "1/1000",
      "start": 0,
      "start_time": "0.000000",
      "end": 0,
      "end_time": "0.000000",
      "tags": {
        "title": "Opening Credits"
      }
    },
    {
      "id": 1,
      "time_base": "1/1000",
      "start": 0,
      "start_time": "0.000000",
      "end": 289413,
      "end_time": "289.413000",
      "tags": {
        "title": "Prologue"
      }
    },
    {
      "id": 2,
      "time_base": "1/10000000",
      "start": 2894130000,
      "start_time": "289.413000",
      "end": 2894130610,
      "end_time": "289.413061",
      "tags": {
        "title": "Chapter 1. So This Is Vietnam"
      }
    }
  ],
```
Thus, after the split, we have the previous chapter with 0 seconds, the current chapter, and the next chapter with less than .1ms . This should not be confused with the metatag in lines 187-217 in which the title is being displayed at line 203 in format → tags → title. Here is a edited extract:
```
  "format": {
    "filename": "/audiobooks/Audiobooks/Non-Fiction/Good to Go - The Life and times of a Decorated Member of the U.S. Navy's Elite Seal Team Two/02 - Good to Go - Prologue.m4b",
    "start_time": "0.000000",
    "duration": "289.413061",
    "size": "2344796",
    "tags": {
      "title": "Prologue",
      "compilation": "0",
      "track": "2/26"
    }
  }
```
Thus, after the split, there should only be ONE chapter in the chapters section, and that chapter title should equal the title in the format → tags → title section. 

[PROBE - 02 - Good to Go - Prologue.txt](https://github.com/user-attachments/files/17147065/PROBE.-.02.-.Good.to.Go.-.Prologue.txt)
[Full OpenAudible Log.zip](https://github.com/user-attachments/files/17147275/Full.OpenAudible.Log.zip)

**Comment by openaudible** [comment]
Great report. Confirmed.
It looks like a strange thing with ffmpeg. I am working on a fix -- but even when I tell ffmpeg to not map any meta data, it still seems to pull chapter meta from the existing full book. Will keep working on it.

**Comment by ZLoth** [comment]
> Great report. Confirmed. It looks like a strange thing with ffmpeg. I am working on a fix -- but even when I tell ffmpeg to not map any meta data, it still seems to pull chapter meta from the existing full book. Will keep working on it.

Thank you. Have fun storming the castle.

**Comment by openaudible** [comment]
Rabbit hole... posted another beta. This should have one chapter but not positive it is exactly what is best. 
https://openaudible.org/beta

**Comment by ZLoth** [comment]
Looks like it's fixed now. What this the solution:
> Ffmpeg has a separate flag for `map_chapters`.
> `-map_chapters -1` to not keep chapters

**Comment by openaudible** [comment]
I just posted a new beta this morning that improves last night one. 

https://openaudible.org/beta

**Comment by openaudible** [comment]
Thank you for the reports. Fixed in 4.4.7. 
https://openaudible.org/

**Comment by ZLoth** [comment]
> Thank you for the reports. Fixed in 4.4.7. https://openaudible.org/

Thank you very much for resolving this issue as I use this tool to get books and upload them to AudioBookShelf.
